core/commands/ipns.go

```diff
@@ -89,15 +96,29 @@ Resolve the value of a reference:
 		}
 
 		var name string
-		if len(req.Arguments()) == 0 {
-			if n.Identity == "" {
-				res.SetError(errors.New("Identity not loaded!"), cmds.ErrNormal)
+
+		if kname, found, _ := req.Option("key").String(); found {
```

Comment:
I think we want to disallow having a key specified and an argument specified at the same time. Either you use `ipfs name resolve -k=foo` or `ipfs name resolve /ipns/QmFooBar`, `ipfs name resolve -k=foo /ipns/QmFooBar` doesnt really make much sense.